lib/engineCustomConditions.js

```diff
@@ -0,0 +1,262 @@
+/* global -Promise */
+'use strict';
+
+var utils = require('./utils'),
+    _ = require('underscore'),
+    Promise = require('bluebird');
+
+var EngineCustomConditions = (function() {
+    return function() {
+
+        /* hmda-syntactical */
```

Comment:
They are just grouping/section type comments, so there still is value, even in the smaller components. Additional jsdoc comments would help, especially documentation tying each function to it's actual edit.